core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/entity/EntitySelectorFactory.java

```diff
@@ -197,15 +198,18 @@ private EntitySelector<Solution_> applyFiltering(EntitySelector<Solution_> entit
         if (hasFiltering(entityDescriptor)) {
             List<SelectionFilter<Solution_, Object>> filterList = new ArrayList<>(config.getFilterClass() == null ? 1 : 2);
             if (config.getFilterClass() != null) {
-                filterList.add(ConfigUtils.newInstance(config, "filterClass", config.getFilterClass()));
+                SelectionFilter<Solution_, Object> selectionFilter =
+                        ConfigUtils.newInstance(config, "filterClass", config.getFilterClass());
+                SelectionFilter<Solution_, Object> wrappedSelectionFilter = new WrappedSelectionFilter<>(selectionFilter);
```

Comment:
Same story. The builder paramater (DescriptorPolicy etc?) could have `Map<Class, instance>` to cache them:
```
SelectionFilter<Solution_, Object> selectionFilter = cacheMap.computeIfAbsent(config.getFilterClass(), c ->
                        ConfigUtils.newInstance(config, "filterClass", c);
```

The trouble is getting that thing that holds that cache here...